File: 997-find-the-town-judge/997-find-the-town-judge.py
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        inDeg = list()
        
        for i in range(n):
            inDeg.append(0)
            
        for i in range(len(trust)):
            inDeg[trust[i][1] - 1]+=1
            inDeg[trust[i][0] - 1]-=1
        
        for i in range(len(inDeg)):
            if inDeg[i] == n-1:
                return i+1
        
        return -1
        

File: 997-find-the-town-judge/997-find-the-town-judge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] inDeg = new int[n];
        
        for(int[] t : trust){
            inDeg[t[1] - 1]++;
            inDeg[t[0] - 1]--;
        }
        
        for(int i=0;i<n;i++){
            if(inDeg[i] == n-1){
                return i+1;
            }
        }
        return -1;
    }
}